automation-tests/test/scratchOrg.ts

```diff
@@ -0,0 +1,328 @@
+  EnvironmentSettings
+} from './environmentSettings';
+import {
+  utilities
+} from './utilities';
+
+const exec = util.promisify(child_process.exec);
+
+export class ScratchOrg {
+  private testSuiteSuffixName: string = undefined;
+  private reuseScratchOrg = false;
+  private projectFolderPath: string = undefined;
+  private prompt: QuickOpenBox | InputBox = undefined;
+  private scratchOrgAliasName: string = undefined;
+
+  public constructor(testSuiteSuffixName: string, reuseScratchOrg: boolean) {
+    this.testSuiteSuffixName = testSuiteSuffixName;
+    this.reuseScratchOrg = reuseScratchOrg;
+
+    // To have all scratch orgs be reused, uncomment the following line:
+    // this.reuseScratchOrg = true;
+  }
+
+  public async setUp(): Promise<void> {
+    await this.setUpTestingEnvironment();
+    await this.createProject();
+    await this.authorizeDevHub();
+    await this.createDefaultScratchOrg();
+  }
+
+  public async tearDown(): Promise<void> {
+    if (this.scratchOrgAliasName && !this.reuseScratchOrg) {
+      // To use VS Code's Terminal view, use:
+      // const workbench = await (await browser.getWorkbench()).wait();
+      // await utilities.executeCommand(workbench, `sfdx force:org:delete -u ${this.scratchOrgAliasName} --noprompt`);
+
+      // The Terminal view can be a bit unreliable, so directly call exec() instead:
+      await exec(`sfdx force:org:delete -u ${this.scratchOrgAliasName} --noprompt`);
+    }
+
+    if (this.projectFolderPath) {
+      await utilities.removeFolder(this.projectFolderPath);
+    }
+  }
+
+  public async setUpTestingEnvironment(): Promise<void> {
+    utilities.log('');
+    utilities.log(`${this.testSuiteSuffixName} - Starting setUpTestingEnvironment()...`);
+
+    const tempFolderPath = path.join(__dirname, '..', 'e2e-temp');
+    this.projectFolderPath = path.join(tempFolderPath, this.tempProjectName);
+
+    // Clean up the temp folder, just in case there are stale files there.
+    if (fs.existsSync(this.projectFolderPath)) {
+      await utilities.removeFolder(this.projectFolderPath);
+      await utilities.pause(1);
+    }
+
+    // Now create the folders.
+    if (!fs.existsSync(tempFolderPath)) {
+      await utilities.createFolder(tempFolderPath);
+      await utilities.pause(1);
+    }
+
+    await utilities.createFolder(this.projectFolderPath);
+    await utilities.pause(1);
+
+    utilities.log(`${this.testSuiteSuffixName} - ...finished setUpTestingEnvironment()`);
+    utilities.log('');
+  }
+
+  public async createProject(): Promise<void> {
+    utilities.log('');
+    utilities.log(`${this.testSuiteSuffixName} - Starting createProject()...`);
+
+    const workbench = await (await browser.getWorkbench()).wait();
+
+    this.prompt = await utilities.executeQuickPick(workbench, 'SFDX: Create Project');
+    // Selecting "SFDX: Create Project" causes the extension to be loaded, and this takes a while.
+    await utilities.pause(10);
+
+    // Select the "Standard" project type.
```

Comment:
I'm loving all the comments in here and how easy it is to follow. That said, each comment + block feels like a discrete function. Would it be worth separating it all to functions to keep readability high and also strip out the comments?

Reply:
Reviewed this and opted to keep our tests DAMP, more readable and easy-breezy, so the comments will stay for flow.